### PR TITLE
Collections component

### PR DESCRIPTION
**Pull request by rlew421**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
After Josh worked on this and got almost all of the way there, we paired and fixed a few errors to get it fully functional.

* Passed user_id into the fetch url by setting it equal to current_user's id
* Fixed the fetch url to be "/api/:user_id/collections" rather than "/api/v0/:user_id/collections"
* Initially defined collectionItems as an empty array
* After API fetch is made, reset state and set collectionItems equal to the array of collections returned from the fetch

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [x] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Comment by dylanconnolly** [comment]
With regards to testing this functionality and since we aren't entirely sure how to test using React, I think we could build out a test or two in the spec/systems/new_collection_spec.rb to make sure collection titles appear on the readinglist page?

File: app/javascript/readingList/readingList.jsx [hunk 1]
```diff
@@ -60,8 +61,24 @@ export class ReadingList extends Component {
...
   }
 
+  getCollections = (user_id) => {
+    const url =`api/${user_id}/collections`
+    console.log('URL!!!! ->', url)
+  fetch(url)
+  .then(response => {
+    return response.json()})
+  .then((data) => {
+    console.log('DATA!!! ->',data)
```
Comment (dylanconnolly) [inline app/javascript/readingList/readingList.jsx]: out of curiousity, are the `console.log` lines simply used to help debug when you were building this or are they needed when making fetch requests? Asking for my own understanding

Comment (jobannon) [reply app/javascript/readingList/readingList.jsx]: just working on the prototype.  They will be removed and are not needed for the fetch request.  We are attempting to see what data we would be getting.

File: app/javascript/readingList/readingList.jsx [hunk 2]
```diff
@@ -243,6 +272,7 @@ export class ReadingList extends Component {
             </div>
             <div>
               {items.length > 0 ? itemsToRender : this.renderEmptyItems()}
+              {collectionItemsToRender }
```
Comment (dylanconnolly) [inline app/javascript/readingList/readingList.jsx]: To get this in its own container I think we can reuse some of the div classes above, so wrap it in like
```
<div className="items-container">
    <div className={`results`}>
            <div className="results-header">
                  Collections
             </div>
             {collectionItemsToRender}
....
```

File: app/javascript/readingList/readingList.jsx (hunk 1 above)
Comment (dylanconnolly) [reply app/javascript/readingList/readingList.jsx]: Cool, got it. Just wanted to have it straight in my head what was going on, thanks for clarifying

**Pull request closed without merging**